### PR TITLE
Add deprecations for future flags

### PR DESCRIPTION
**Pull request by brophdawg11**

### CODE REVIEW

File: .changeset/rude-clocks-prove.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@remix-run/dev": minor
+---
+
+Log deprecation warnings for v3 future flags
```
Comment (brophdawg11) [inline .changeset/rude-clocks-prove.md:5]:
```suggestion
- Log deprecation warnings for v3 future flags
- Add `@deprecated` annotations to `json`/`defer` utilities
```

**Comment by wouterds** [comment]
How does this work in combination with
>⚠️ REMIX FUTURE CHANGE: Externally-accessed resource routes will no longer be able to return raw JavaScript objects or `null` in React Router v7 when Single Fetch becomes the default. You can prepare for this change at your convenience by wrapping the data returned from your `action` function in the `routes/users` route with `json()`.  For instructions on making this change, see https://remix.run/docs/en/v2.13.1/guides/single-fetch#resource-routes

**Comment by tmcw** [comment]
There's quite a lot of documentation that still includes the `json()` method - the [documentation for loader itself](https://remix.run/docs/en/main/route/loader) uses `json()`, as well as [the tutorials](https://remix.run/docs/en/main/start/tutorial#updating-data). Folks who start from those directions will get warnings from Remix, which is not a very good onboarding experience.

**Comment by danielvanc** [comment]
> > There's quite a lot of documentation that still includes the `json()` method  ... Folks who start from those directions will get warnings from Remix, which is not a very good onboarding experience.
> 
> Not to mention there is nothing I can find pointing me to what I should do instead? It would be nice to also have the deprecation message point us to an alternative.

Yeah i agree. it should at least show examples of links to pages like these:
https://reactrouter.com/dev/start/framework/data-loading

**Comment by bhouston** [comment]
I recently adopted Remix and I am noticing that my code has deprecation warnings.  It appears to be related to this issue.

The issue is that I am using json() in order to return data and a status code.  What is the replacement code for this?

**Comment by peter-skillmasters** [comment]
The return type of `Response.json()` is just `Response`, which means I lose type information when I use it. So if you need to apply a custom status code or use `commitSession` with a JSON body, it's a pretty major DX downgrade from `json()`. What's the recommendation here? 

```ts
function oldLoader() {
  return json({ foo: "bar"}, {status: 201})
}

type OldLoaderReturn = ReturnType<typeof useLoaderData<typeof oldLoader>>
// => { foo: string }

function newLoader() {
  return Response.json({ foo: "bar"}, {status: 201})
}

type NewLoaderReturn = ReturnType<typeof useLoaderData<typeof newLoader>>
// => any
```

**Comment by peter-skillmasters** [comment]
> If you're using single fetch, I believe this is what you're looking for: https://remix.run/docs/en/main/utils/data

Thank you! That's very helpful. The deprecation message should probably point to this as an alternative

**Pull request merged**